### ISSUE TITLE
How to padding in data iterator

### ISSUE THREAD

**Issue by playma** [allenai/allennlp#2984]
I have two inputs (term1 and term2).
How can I get the same padding lengh of term1 and term2 in data iterator?

In each batch, term1 padding with the longest length in term1,  term2 padding with the longest length in term2.
But I want to get term1 and term2 padding with the  longest length in both term1 and term2
They are indexed by bert, so there are token, tokens-offsets, tokens-type-ids in the dict.

Can any one help me? Thx

```
iterator = BucketIterator(batch_size=30, sorting_keys=[("term1", "num_tokens"), ("term2", "num_tokens")])
# iterator = BasicIterator(batch_size=30)
iterator.index_with(vocab)

for data in tqdm(iter(iterator(dataset, num_epochs=1))):
    data = move_to_device(data, cuda_device=cuda_device)

    term1, term2, label = data['term1'], data['term2'], data['label']
```

**Comment by DeNeutoy** [comment]
Hi,

Broadly that is not possible. However, it shouldn't be necessary - could you describe a bit what you are trying to achieve on a broader scale? I might be able to suggest a different approach.

**Comment by brendan-ai2** [comment]
@playma , we're closing this due to inactivity, but feel free to comment again if you pick it back up.

**Comment by playma** [comment]
@DeNeutoy I want to use bert to extract the features and concat features of term1 and term2 like siamese model.
